app/controllers/jobs_controller.rb

```diff
@@ -5,40 +5,80 @@ def index
   end
 
   def new
-    @company = Company.find(params[:company_id])
-    @job = Job.new()
+    @categories = Category.all
+    @company    = Company.find(params[:company_id])
+    @job        = Job.new()
   end
 
   def create
-    @company = Company.find(params[:company_id])
-    @job = @company.jobs.new(job_params)
+    @categories = Category.all
+    @company    = Company.find(params[:company_id])
+    @job        = @company.jobs.new(job_params)
     if @job.save
       flash[:success] = "You created #{@job.title} at #{@company.name}"
       redirect_to company_job_path(@company, @job)
     else
+      @error = "A job must have a Title, level of interest, city, and category!"
       render :new
     end
   end
 
   def show
-    @job = Job.find(params[:id])
+    @job      = Job.find(params[:id])
+    @company  = Company.find(params[:company_id])
+    @category = category
+    @comments = @job.comments.all
```

Comment:
no need to call `all` on the end of this - `@job.comments` will return to you all the comments for that job!